File: download_picsum.py
import os
import urllib.request
import ssl
import time

# Отключаем проверку SSL (часто помогает при проблемах с сетью)
ssl._create_default_https_context = ssl._create_unverified_context

# Список категорий и ID картинок с Picsum (высокое качество, тематические)
categories = [
    ('transport', 'https://picsum.photos/id/1071/400/400'),
    ('real-estate', 'https://picsum.photos/id/1031/400/400'),
    ('jobs', 'https://picsum.photos/id/1/400/400'),
    ('services', 'https://picsum.photos/id/1070/400/400'),
    ('electronics', 'https://picsum.photos/id/367/400/400'),
    ('home', 'https://picsum.photos/id/1062/400/400'),
    ('clothing', 'https://picsum.photos/id/1059/400/400'),
    ('parts', 'https://picsum.photos/id/252/400/400'),
    ('hobby', 'https://picsum.photos/id/96/400/400'),
    ('pets', 'https://picsum.photos/id/237/400/400'),
    ('beauty', 'https://picsum.photos/id/360/400/400'),
    ('kids', 'https://picsum.photos/id/1084/400/400')
]

save_dir = os.path.join("public", "categories")
if not os.path.exists(save_dir):
    os.makedirs(save_dir)

print(f"Start downloading to {save_dir}...", flush=True)

opener = urllib.request.build_opener()
opener.addheaders = [('User-Agent', 'Mozilla/5.0')]
urllib.request.install_opener(opener)

for name, url in categories:
    try:
        filename = os.path.join(save_dir, f"{name}.jpg")
        print(f"Downloading {name}...", end=" ", flush=True)
        urllib.request.urlretrieve(url, filename)
        print("OK", flush=True)
    except Exception as e:
        print(f"ERROR: {e}", flush=True)
    time.sleep(0.5)

print("\nAll downloads finished!", flush=True)

File: download_picsum.js
const fs = require('fs');
const https = require('https');
const path = require('path');

const categories = [
    { name: 'transport', url: 'https://picsum.photos/id/1071/400/400' },
    { name: 'real-estate', url: 'https://picsum.photos/id/1031/400/400' },
    { name: 'jobs', url: 'https://picsum.photos/id/1/400/400' },
    { name: 'services', url: 'https://picsum.photos/id/1070/400/400' },
    { name: 'electronics', url: 'https://picsum.photos/id/367/400/400' },
    { name: 'home', url: 'https://picsum.photos/id/1062/400/400' },
    { name: 'clothing', url: 'https://picsum.photos/id/1059/400/400' },
    { name: 'parts', url: 'https://picsum.photos/id/252/400/400' },
    { name: 'hobby', url: 'https://picsum.photos/id/96/400/400' },
    { name: 'pets', url: 'https://picsum.photos/id/237/400/400' },
    { name: 'beauty', url: 'https://picsum.photos/id/360/400/400' },
    { name: 'kids', url: 'https://picsum.photos/id/1084/400/400' }
];

const downloadDir = path.join(__dirname, 'public', 'categories');
if (!fs.existsSync(downloadDir)) fs.mkdirSync(downloadDir, { recursive: true });

async function download(url, dest) {
    return new Promise((resolve, reject) => {
        const file = fs.createWriteStream(dest);
        https.get(url, { rejectUnauthorized: false }, (response) => {
            if (response.statusCode >= 300 && response.statusCode < 400 && response.headers.location) {
                // Handle Redirects manually often needed for picsum
                download(response.headers.location, dest).then(resolve).catch(reject);
                return;
            }
            response.pipe(file);
            file.on('finish', () => {
                file.close(resolve);
            });
        }).on('error', (err) => {
            fs.unlink(dest, () => { });
            reject(err);
        });
    });
}

(async () => {
    console.log('Starting Node download...');
    for (const cat of categories) {
        try {
            console.log(`Downloading ${cat.name}...`);
            await download(cat.url, path.join(downloadDir, `${cat.name}.jpg`));
            console.log(`Saved ${cat.name}.jpg`);
        } catch (e) {
            console.error(`Error ${cat.name}:`, e.message);
        }
    }
    console.log('Done.');
})();
